### DISCUSSION TITLE
aruco linux

### DISCUSSION THREAD

**Discussion by lolmuele** [opentrack/opentrack#1435] [Q&A] [answered]
Can anybody explain how the i could get opentrack with aruco on linux.
I have only found one guide in the depths of the internet and it didn't work at all for me.
I can install opentrack, but have no idea how to add aruco although there are folders like aruco-tracker and aruco inside the opentrack folder.
i am running ubuntu 20.04 and am relatively new to linux (only had it for abt 8 months)

**Comment by sthalik** [comment, marked as answer]
Build https://github.com/opentrack/aruco. Set `SDK_ARUCO_LIBPATH`.

**Reply by lolmuele** [reply]
where do i set the libpath ?

**Comment by sthalik** [comment]
cmake cache

**Reply by lolmuele** [reply]
I did this 
//Aruco paper marker tracker static library path
SDK_ARUCO_LIBPATH:FILEPATH=/usr/local/lib/libaruco.a
but it still isnt there do i have to rebuild opentrack for it to appear ?

**Reply by sthalik** [reply]
Yes.

**Reply by lolmuele** [reply]
I get this error for some reason (i am using the opentrack aruco version )

Building CXX object CMakeFiles/cmTC_9dc7e.dir/abi.cpp.o

  /usr/bin/c++ -I/usr/local/lib/include -ggdb -Wall -Wextra -Wpedantic -fPIC
  -std=c++17 -o CMakeFiles/cmTC_9dc7e.dir/abi.cpp.o -c
  /home/lolmuele/opentrack/tracker-aruco/compile-test/abi.cpp

  /home/lolmuele/opentrack/tracker-aruco/compile-test/abi.cpp:1:10: fatal
  error: aruco/abi.h: No such file or directory

      1 | #include <aruco/abi.h>
        |          ^~~~~~~~~~~~~

  compilation terminated.

  make[1]: *** [CMakeFiles/cmTC_9dc7e.dir/build.make:66:
  CMakeFiles/cmTC_9dc7e.dir/abi.cpp.o] Error 1

  make[1]: Leaving directory
  '/home/lolmuele/opentrack/tracker-aruco/CMakeFiles/CMakeTmp'

  make: *** [Makefile:121: cmTC_9dc7e/fast] Error 2

  

  

  Must use Aruco fork from <https://github.com/opentrack/aruco>


-- Configuring incomplete, errors occurred!
See also "/home/lolmuele/opentrack/CMakeFiles/CMakeOutput.log".

**Reply by lolmuele** [reply]
no, even rebuilding on entirely new source doesn't seem to work i have no ideas

**Reply by lolmuele** [reply]
i was just being stupid and used a temporary version of libaruco my system made it works now thx

**Comment by sthalik** [comment]
The install target is broken. I should remove it completely. Point to the `.a` inside the build directory instead.

**Reply by sthalik** [reply]
And done.
